source/common/http/conn_manager_impl.cc

```diff
@@ -502,6 +502,8 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(HeaderMapPtr&& headers,
     state_.saw_connection_close_ = true;
   }
 
+  const HeaderEntry* decorator_operation = request_headers_->EnvoyDecoratorOperation();
```

Comment:
So at minimum, I think you are grabbing a header before headers are cleaned, which means this can get deleted. Can you remove the local variable and do this inline below? (Or just pass the full constant header map ref to the trace function).

Reply:
I'll give it a go tomorrow - although I think the problem is that the `x-envoy-decorator-operation` is being removed in the sanitize step, as previously when I commented this line out the tests/coverage worked.